apps/ckit-app/src/views/Assets/index.tsx

```diff
@@ -3,20 +3,20 @@ import { observer } from 'mobx-react-lite';
 import React from 'react';
 import { AssetList } from 'components/Assets';
 import { StyledCardWrapper } from 'components/Styled';
-import { WalletContainer } from 'containers/WalletContainer';
+import { useSigner, WalletContainer } from 'containers/WalletContainer';
 
 export const AssetsView: React.FC = observer(() => {
   const { selectedWallet } = WalletContainer.useContainer();
-  const showAssetList = selectedWallet?.signer;
+  const { address: signerAddress } = useSigner(selectedWallet?.signer);
```

Comment:
I found there was may `useSigner(selectedWallet?.signer)` or `if(ckitProvider) { ... }`, maybe we can refactor them in a simply way like this

```tsx
function useSigner(): Signer {
  const signerState = SignerContainer.useContainer()
  if(!signerState.signer) boom('The signer is not found');

  return signerState;
}


const SignerEntry: React.FC = ({ children, fallback }) => {
   const signerState = SignerContainer.useContainer()

  if(!signerState) return  fallback;
  return children
}

// and then we use it like this
<QueryProvider>
  <CkitProvider>
    <XProvider>
        <SignerEntry>
              <SomeComponent />
      </SignerEntry>
       

```

Reply:
impl it in the future pr